tests/cephfs/cephfs_pool_pg_size_change.py

```diff
@@ -0,0 +1,202 @@
+        log.info("Checking for heartbeat map timeout issue")
+        rc = fs_util.heartbeat_map(mdss[0])
+        if rc == 1:
+            log.error("heartbeat map timeout issue found")
+            return 1
+        log.info(
+            "Changing cephfs data and metadata pool pg_num and pgp_num to existing size '-1' with client IO running"
+        )
+        no_of_files = "{1..1000}"
+        data_pool_pg_num = str(int(data_pool_pg_num) - 1)
+        metadata_pool_pg_num = str(int(metadata_pool_pg_num) - 1)
+        commands = [
+            f'for n in {no_of_files}; do     dd if=/dev/urandom of={kernel_mount_dir}/dir3/file$( printf %03d "$n" )'
+            f" bs=1M count=10; done",
+            f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 4 "
+            f"--files 1000 --files-per-dir 10 --top {kernel_mount_dir}/dir4",
+            f"ceph osd pool set {data_pool} pg_num {data_pool_pg_num}",
+            f"ceph osd pool set {data_pool} pgp_num {data_pool_pg_num}",
+            f"ceph osd pool set {metadata_pool} pg_num {metadata_pool_pg_num}",
+            f"ceph osd pool set {metadata_pool} pgp_num {metadata_pool_pg_num}",
+        ]
+        with parallel() as p:
+            for num in range(0, 6):
+                p.spawn(clients[0].exec_command, sudo=True, cmd=commands[num])
+                time.sleep(1)
+        log.info("Verifying pgs are in active+clean state")
+        rc = check_clean_pgs(clients)
+        if rc == 1:
+            return 1
+        log.info(
+            "Change cephfs data and metadata pool pg_num and pgp_num to existing size '+1' with client IO running"
+        )
+        data_pool_pg_num = str(int(data_pool_pg_num) + 1)
+        metadata_pool_pg_num = str(int(metadata_pool_pg_num) + 1)
+        commands = [
+            f'for n in {no_of_files}; do     dd if=/dev/urandom of={kernel_mount_dir}/dir5/file$( printf %03d "$n" )'
+            f" bs=1M count=10; done",
+            f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 4 "
+            f"--files 1000 --files-per-dir 10 --top {kernel_mount_dir}/dir6",
+            f"ceph osd pool set {data_pool} pg_num {data_pool_pg_num}",
+            f"ceph osd pool set {data_pool} pgp_num {data_pool_pg_num}",
+            f"ceph osd pool set {metadata_pool} pg_num {metadata_pool_pg_num}",
+            f"ceph osd pool set {metadata_pool} pgp_num {metadata_pool_pg_num}",
+        ]
+        with parallel() as p:
+            for num in range(0, 6):
+                p.spawn(clients[0].exec_command, sudo=True, cmd=commands[num])
+                time.sleep(1)
+        log.info("Verifying pgs are in active+clean state")
+        rc = check_clean_pgs(clients)
+        if rc == 1:
+            return 1
+        log.info("Cleaning up the system")
+        out, rc = clients[0].exec_command(sudo=True, cmd=f"rm -rf {mount_points[1]}/*")
```

Comment:
If possible can we put this in finally block?
So that other test cases will not have any much impact if any